odxtools/diaglayer.py

```diff
@@ -751,15 +753,16 @@ def __str__(self) -> str:
 
 
 def read_diag_layer_container_from_odx(et_element, enable_candela_workarounds=True):
-    short_name = et_element.find("SHORT-NAME").text
+    short_name = et_element.findtext("SHORT-NAME")
+    assert short_name is not None
     long_name = et_element.findtext("LONG-NAME")
 
     # create the current ODX "document fragment" (description of the
     # current document for references and IDs)
     doc_frags = [ OdxDocFragment(short_name, "CONTAINER") ]
 
-    id = OdxLinkId.from_et(et_element, doc_frags)
-
+    odx_id = OdxLinkId.from_et(et_element, doc_frags)
+    assert odx_id is not None
```

Comment:
Are you sure ID is required here?

Reply:
after checking with the XSD, I am: DIAG-LAYER-CONTAINER is an extension of ODX-CATEGORY which in turn extends ELEMENT-ID..